cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go

```diff
@@ -0,0 +1,263 @@
+	ref, err := GceRefFromProviderId(node.Spec.ProviderID)
+	if err != nil {
+		return nil, err
+	}
+	mig, err := gce.gceManager.GetMigForInstance(ref)
+	return mig, err
+}
+
+// GceRef contains s reference to some entity in GCE/GKE world.
+type GceRef struct {
+	Project string
+	Zone    string
+	Name    string
+}
+
+// GceRefFromProviderId creates InstanceConfig object
+// from provider id which must be in format:
+// gce://<project-id>/<zone>/<name>
+// TODO(piosz): add better check whether the id is correct
+func GceRefFromProviderId(id string) (*GceRef, error) {
+	splitted := strings.Split(id[6:], "/")
+	if len(splitted) != 3 {
+		return nil, fmt.Errorf("Wrong id: expected format gce://<project-id>/<zone>/<name>, got %v", id)
+	}
+	return &GceRef{
+		Project: splitted[0],
+		Zone:    splitted[1],
+		Name:    splitted[2],
+	}, nil
+}
+
+// Mig implements NodeGroup interfrace.
+type Mig struct {
+	GceRef
+
+	gceManager *GceManager
+
+	minSize    int
+	maxSize    int
+	sampleNode *kube_api.Node
+}
+
+// MaxSize returns maximum size of the node group.
+func (mig *Mig) MaxSize() int {
+	return mig.maxSize
+}
+
+// MinSize returns minimum size of the node group.
+func (mig *Mig) MinSize() int {
+	return mig.minSize
+}
+
+// Size returns the current TARGET size of the node group. It is possible that the
+// number is different from the number of nodes registered in Kuberentes.
+func (mig *Mig) Size() (int, error) {
+	size, err := mig.gceManager.GetMigSize(mig)
+	return int(size), err
+}
+
+// SampleNode returns a sample node for the mig. Assumes that mig definition doesn't change over time.
+// The node may not exist anymore.
+func (mig *Mig) SampleNode() (*kube_api.Node, error) {
+	if mig.sampleNode != nil {
+		return mig.sampleNode, nil
+	}
+	return nil, fmt.Errorf("no sample node available")
+}
+
+// IncreaseSize increases Mig size
+func (mig *Mig) IncreaseSize(delta int) error {
+	if delta <= 0 {
+		return fmt.Errorf("size increase must be positive")
+	}
+	size, err := mig.gceManager.GetMigSize(mig)
+	if err != nil {
+		return err
+	}
+	if int(size)+delta > mig.MaxSize() {
+		return fmt.Errorf("size increase to large - desired:%d max:%d", int(size)+delta, mig.MaxSize())
+	}
+	return mig.gceManager.SetMigSize(mig, size+int64(delta))
+}
+
+// Belongs retruns true if the given node belongs to the NodeGroup.
+func (mig *Mig) Belongs(node *kube_api.Node) (bool, error) {
+	ref, err := GceRefFromProviderId(node.Spec.ProviderID)
+	if err != nil {
+		return false, err
+	}
+	targetMig, err := mig.gceManager.GetMigForInstance(ref)
+	if err != nil {
+		return false, err
+	}
+	if targetMig == nil {
+		return false, fmt.Errorf("%s doesn't belong to a known mig", node.Name)
+	}
+	if targetMig.Id() != mig.Id() {
+		return false, nil
+	}
+	return true, nil
+}
+
+// DeleteNode deletes the node from the group.
+func (mig *Mig) DeleteNode(node *kube_api.Node) error {
```

Comment:
DeleteNodes? As for now we only support scaling down by 1 node, but probably we want to change it soon